src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GCStaticsNode.cs

```diff
@@ -12,11 +14,34 @@ namespace ILCompiler.DependencyAnalysis
     public class GCStaticsNode : ObjectNode, IExportableSymbolNode
     {
         private MetadataType _type;
+        private List<FieldDesc> _preinitFields;
 
         public GCStaticsNode(MetadataType type)
         {
             Debug.Assert(!type.IsCanonicalSubtype(CanonicalFormKind.Specific));
             _type = type;
+
+            // Go through and find all fields with preinitialized data
+            foreach (var field in _type.GetFields())
```

Comment:
Why can't `GCStaticsPreInitDataNode` go over the field list and sort it? It's rather odd to do sorting here even though it's a different class that actually needs the sorted field list...